youtube_dlc/downloader/youtube_live_chat.py

```diff
@@ -43,23 +47,35 @@ def parse_yt_initial_data(data):
                 except AttributeError:
                     continue
 
-        def download_and_parse_fragment(url, frag_index):
+        def parse_ytcfg(data):
+            data = data.decode('utf-8', 'replace')
+            try:
+                raw_json = re.search(r'ytcfg\.set\s*\(\s*({.+?})\s*\)\s*;', data).group(1)
+                return json.loads(raw_json)
+            except IndexError:
+                return None
+
+        def download_and_parse_fragment(url, frag_index, request_data):
             count = 0
             while count <= fragment_retries:
                 try:
-                    success, raw_fragment = dl_fragment(url)
+                    success, raw_fragment = dl_fragment(url, request_data, {'content-type': 'application/json'})
                     if not success:
                         return False, None, None
                     data = parse_yt_initial_data(raw_fragment)
                     if not data:
+                        data = {}
                         raw_data = json.loads(raw_fragment)
                         # sometimes youtube replies with a list
                         if not isinstance(raw_data, list):
                             raw_data = [raw_data]
-                        try:
-                            data = next(item['response'] for item in raw_data if 'response' in item)
-                        except StopIteration:
-                            data = {}
+                        for item in raw_data:
+                            # data is sometimes behind 'response'
+                            if 'response' in item:
+                                data = item['response']
+                            else:
+                                data = item
+                            break
```

Comment:
Are we sure that the data we want is always in the first item? If so, this can be simplified to:
```suggestion
                        data = try_get(raw_data, lambda x: x[0].get('response', x[0])) or {}
```

If not, some check must be done to ensure this is the data we want before breaking
```suggestion
                        for item in raw_data:
                            data = item.get('response', item)
                            if some_check():
                                break
```

Reply:
I don't remember seeing the GET resource outputting an array, so it was probably added to address some bug afterwards. Maybe the array check could be removed completely to simplify even further.

Reply:
I added the array. A week or so ago, youtube was sometimes returning an array. I had to look for the element in the array that has the `reponse` key. Hence

```
                        # sometimes youtube replies with a list
                        if not isinstance(raw_data, list):
                            raw_data = [raw_data]
                        try:
                            data = next(item['response'] for item in raw_data if 'response' in item)
```

But if there is no array in the POST reply, then you can change it

Reply:
See 4d608b522f7f05c7dbc4b74e6183e93cc95d6a0c

Reply:
Right, thanks for the explanation. I removed the check and the code is now simpler